sale.py

```diff
@@ -222,6 +222,12 @@ def create(cls, vlist):
             sale.check_create_access()
         return sales
 
-    # TODO: On copying an order from a channel the user does not have
-    # create access, default to the current channel of the user. If there
-    # is no current channel, blow up
+    @classmethod
+    def copy(cls, sales, default=None):
+        """
+        Duplicating records
+        """
+        if default is None:
+            default = {}
+        default['channel'] = cls.default_channel()
```

Comment:
If user does not have create access to this channel then only fetch default_channel